### PR TITLE
CXX-3150 Post-release changes for 4.0.0

### PR DESCRIPTION
**Pull request by eramongodb**
Followup changes after the 4.0.0 release.

- The static analysis report is renamed to `static_analysis_report-X.Y.Z` for consistency with prior reports.
- The "Driver status by family and version" section is renamed to "Driver Development Status" and its contents simplified to list only what is necessary.
    - The note regarding the meaning of "stability" is removed in favor of a more straightforward "Development Stability" column and a "Do not use in production!" notice for the `master` branch.
    - A new "ABI Stability" column is included in preparation of upcoming ABI v1 interfaces.
- The list of documentation versions in the API homepage is reformatted to better utilize screenspace.
- Hugo apparently appends a trailing `/` in the substitution of `baseUrl`, defeating the "fixes" in https://github.com/mongodb/mongo-cxx-driver/pull/1238. (Something changed during Hugo version 0.92.2 -> 0.123.7?) The trailing `/` added in https://github.com/mongodb/mongo-cxx-driver/pull/1071 is restored (since it better matches the actual expansion) and appropriate changes made in usage sites.
- https://github.com/mongodb/mongo-cxx-driver/pull/1239 overlooked the required addition of an `--exclude` flag to the Hugo `rsync --delete` command to avoid the sitemap index file from being deleted. The flag is added, as well as a flag to exclude of the new `mongocxx-v4` symlink in the Doxygen rsync command.
- The README contents are reorganized and updated with the new version table format and the latest conversion of `releases/stable` from a branch to a tag.

### CODE REVIEW

File: etc/apidocmenu.md [hunk 1]
```diff
@@ -1,71 +1,26 @@
...
-Currently, no drivers guarantee API or ABI stability.
+[4.0.0](../mongocxx-4.0.0) | [3.11.0](../mongocxx-3.11.0) | [3.10.2](../mongocxx-3.10.2) | [3.10.1](../mongocxx-3.10.1) | [3.10.0](../mongocxx-3.10.0) | [3.9.0](../mongocxx-3.9.0) | [3.8.1](../mongocxx-3.8.1) | [3.8.0](../mongocxx-3.8.0) | [3.7.2](../mongocxx-3.7.2) | [3.7.1](../mongocxx-3.7.1) | [3.7.0](../mongocxx-3.7.0) | [3.6.7](../mongocxx-3.6.7) | [3.6.6](../mongocxx-3.6.6) | [3.6.5](../mongocxx-3.6.5) | [3.6.4](../mongocxx-3.6.4) | [3.6.3](../mongocxx-3.6.3) | [3.6.2](../mongocxx-3.6.2) | [3.6.1](../mongocxx-3.6.1) | [3.6.0](../mongocxx-3.6.0) | [3.5.1](../mongocxx-3.5.1) | [3.5.0](../mongocxx-3.5.0) | [3.4.2](../mongocxx-3.4.2) | [3.4.1](../mongocxx-3.4.1) | [3.4.0](../mongocxx-3.4.0) | [3.3.2](../mongocxx-3.3.2) | [3.3.1](../mongocxx-3.3.1) | [3.3.0](../mongocxx-3.3.0) | [3.2.1](../mongocxx-3.2.1) | [3.2.0](../mongocxx-3.2.0) | [3.1.4](../mongocxx-3.1.4/) | [3.1.3](../mongocxx-3.1.3/) | [3.1.2](../mongocxx-3.1.2/) | [3.1.1](../mongocxx-3.1.1/) | [3.1.0](../mongocxx-3.1.0/) | [3.0.3](../mongocxx-3.0.3/) | [3.0.2](../mongocxx-3.0.2/) | [3.0.1](../mongocxx-3.0.1/) | [3.0.0](../mongocxx-3.0.0/)
```
Comment (kevinAlbs) [inline etc/apidocmenu.md:5]: Suggest updating release instructions to account for change of table to list:

> Update the tables in `etc/apidocmenu.md`

To:

> Update the listings in `etc/apidocmenu.md`

File: README.md [hunk 2]
```diff
@@ -6,16 +6,38 @@
...
-is not stable for use in production.
+The `releases/stable` tag indicates the latest stable release.
+This tag corresponds to the driver version which we currently recommend for use in production.
+By default, new users should clone or download the driver version which is currently indicated by this tag.
 
-See [Driver Status by family and version](#driver-status-by-family-and-version)
-for more details about the various versions of the driver.
+```bash
+# Use the latest stable release which is currently recommended for use.
+git clone -b release/stable https://github.com/mongodb/mongo-cxx-driver.git
```
Comment (kevinAlbs) [inline README.md:17]:
```suggestion
git clone -b releases/stable https://github.com/mongodb/mongo-cxx-driver.git
```

**Review by kevinAlbs — APPROVED** [review]
LGTM with minor fixes. I like the simplified compatibility table and version listing.

File: etc/apidocmenu.md (hunk 1 above)
Comment (eramongodb) [reply etc/apidocmenu.md:5]: Updated wording to reference each list/table which needs entries to be updated.

**Pull request merged**